rust/src/container.rs

```diff
@@ -387,6 +398,20 @@ pub fn container_encapsulate(args: Vec<String>) -> CxxResult<()> {
         })
         .collect::<Result<_>>()?;
 
+    let package_structure = match &opt.ocimeta_json {
+        Some(ocimeta_json) => {
+            let path = std::path::Path::new(ocimeta_json);
+            if path.is_file() {
+                let manifest = oci_spec::image::ImageManifest::from_file(path)
+                    .expect("parse ImageManifest from ocimeta");
+                Some(manifest)
+            } else {
+                None
+            }
+        }
+        None => None,
```

Comment:
In this code we're converting `Option<T>` to `Option<U>`, which is `Option::map()`.  However if we use map we need to deal with the fact that we end up with `Option<Result<Manifest>>`.  The `transpose()` method converts that to `Result<Option<Manifest>>`.  So something like this should work:

```
   let parsed_manifest = opt.ocimeta_json.as_ref().map(|p| oci_spec::image::ImageManifest::from_file(&p)).transpose()?;
```

(combining previous suggestions too)